src/Microsoft.Restier.EntityFramework/Model/ModelProducer.cs

```diff
@@ -336,11 +336,46 @@ private static IEdmPrimitiveTypeReference GetPrimitiveTypeReference(EdmProperty
                         efProperty.IsUnicode,
                         efProperty.Nullable);
                 case EdmPrimitiveTypeKind.DateTimeOffset:
-                case EdmPrimitiveTypeKind.Duration:
+                    if (efProperty.PrimitiveType.PrimitiveTypeKind == PrimitiveTypeKind.DateTime)
+                    {
+                        MetadataProperty metadata;
+                        efProperty.MetadataProperties.TryGetValue("Configuration", true, out metadata);
+                        if (metadata != null)
+                        {
+                            object config = metadata.Value;
+                            var columnType = (string)config.GetType().GetProperty("ColumnType").GetValue(config);
+
+                            if (string.Equals(columnType, "date", StringComparison.OrdinalIgnoreCase))
+                            {
+                                return EdmCoreModel.Instance.GetDate(efProperty.Nullable);
+                            }
+                        }
+                    }
```

Comment:
No this works only for EF6, EF7 has a much cleaner approach.
EF7 shares most code files from EF6, but has dedicated ModelProducer and ChangeSetPreparer.